completions/bash/multipass

```diff
@@ -21,25 +21,105 @@ _multipass_complete()
         local cmd="multipass list --format=csv --no-ipv4"
         [ -n "$state" ] && cmd="$cmd | \awk -F',' '\$2 == \"$state\"'"
 
-        local instances=$( \eval $cmd | \grep -Ev '(\+--|Name)' | \cut -d',' -f 1 )
+        local retval=$( \eval $cmd | \grep -Ev '(\+--|Name)' | \cut -d',' -f 1 )
+
+        for instance in $retval; do
+            instances="${instances} ${instance}"
+        done
+    }
+
+    _multipass_snapshots()
+    {
+        local instance="${cur%%.*}"
+        local cmd="multipass info --snapshot-overview "${instance}" --format=csv 2>/dev/null"
+        local retval=$( \eval $cmd | \grep -Ev '(\+--|Instance)' | \cut -d',' -f 2 )
+
+        for snapshot in $retval; do
+            snapshots="${snapshots} ${instance}.${snapshot}"
+        done
+    }
+
+    _multipass_instances_and_snapshots()
+    {
+        if [[ "${cur}" =~ '.' ]]; then
+            _multipass_snapshots
+            _multipass_filter_array "snapshots"
+        fi
+
+        _multipass_instances
+        _multipass_filter_array "instances"
+    }
+
+    _multipass_filter_array()
```

Comment:
What is the purpose of this function? Is this meant only to remove repetitions? If so, isn't that now being handled generically by #3156 ?

Reply:
Yes, I found a way to generalize it and the same code is also being used there. What would be the best way to get the code here? Merge the other PR and rebase on main?

Reply:
Yes, I think that is the best. To unblock this one you can create a temporary base branch out of snapshots that merges the unduplicating branch in, then rebase this on top of that, and turn it into a draft, with a note that it needs the other one to go in. That would allow you to deal with any conflicts early and otherwise progress here. Once the other one is merged, we'd rebase snapshots on main, rebase this on snapshots, and delete the temporary branch.